x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Page.java

```diff
@@ -235,12 +235,8 @@ public void releaseBlocks() {
 
         blocksReleased = true;
 
-        // blocks can be used as multiple columns
-        var map = new IdentityHashMap<Block, Boolean>(mapSize(blocks.length));
         for (Block b : blocks) {
-            if (map.putIfAbsent(b, Boolean.TRUE) == null) {
-                Releasables.closeExpectNoException(b);
-            }
+            Releasables.closeExpectNoException(b);
```

Comment:
This becomes simpler to reason about and doesn't need a hashmap on every release.